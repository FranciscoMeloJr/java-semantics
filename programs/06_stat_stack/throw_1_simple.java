//Testing simple throw.
//Also catch argument shadowing a local variable with the same name.

class Exception {
  Exception(){}
}

public class throw_1_simple {
  public static void main(String[] args) {
    int e = 100;
    try {
      int x = 5;
      if (x >= 0)
        throw new Exception();
      System.out.print("unreachable");
    } catch(Exception e) {
      System.out.println(e+ " ");   // should print Exception
    }
    System.out.println(e+ " ");     // should print 100
    System.out.println("Done!");
  }
}
// Exception
// 100
// Done!
