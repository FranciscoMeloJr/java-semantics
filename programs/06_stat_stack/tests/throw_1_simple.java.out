Exception 
100 
Done!
